features/decks/ui/src/commonMain/kotlin/app.deckbox.ui.decks/builder/composables/DeckBuilderBottomSheet.kt

```diff
@@ -41,6 +48,35 @@ internal fun ColumnScope.DeckBuilderBottomSheet(
   var descriptionValue by remember(state.session.deckOrNull() != null) {
     mutableStateOf(TextFieldValue(state.session.deckOrNull()?.description ?: ""))
   }
+  var nameValue by remember(state.session.deckOrNull() != null) {
+    mutableStateOf(TextFieldValue(state.session.deckOrNull()?.name ?: ""))
+  }
+  BuilderTextField(
+    icon = { Icon(Icons.Rounded.ShortText, contentDescription = null) },
+  ) {
+    TextField(
+      value = nameValue,
+      onValueChange = { value ->
+        nameValue = value
+        eventSink(DeckBuilderUiEvent.EditName(value.text))
+      },
+      placeholder = {
+        Text("Name")
+      },
+      label = {
+        Text("Name")
+      },
+      keyboardOptions = KeyboardOptions(
+        keyboardType = KeyboardType.Text,
+        imeAction = ImeAction.Next,
+      ),
+      singleLine = true,
+      modifier = Modifier
+        .fillMaxWidth()
+        .focusRequester(focusRequester)
+    )
+  }
+  Spacer(Modifier.height(16.dp))
```

Comment:
So part of the `CardBuilder` design was to centralize the duplicate elements between the `DeckBuilder` and `BoosterPackBuilder` UI's so that they would only need to be edited in once place. By hoisting this element out of `CardBuilder` its now duplicated work along with exposing implementation details (i.e. the `FocusRequester`).

I see that this was done because the current setup couldn't account for whether or not the state has loaded for either of these screens (i.e. the `state.session.*orNull() != null` checks). 

What we should do instead here is hoist the `TextFieldValue` out of `CardBuilder` into both `BoosterPackBuilder` and `DeckBuilder` instead. 

e.g. 

**DeckBuilder**  
```kotlin
var deckName by remember(state.session.deckOrNull() != null) {
  mutableStateOf(TextFieldValue(state.session.deckOrNull()?.name ?: ""))
}

CardBuilder(
    name = deckName,
    //...
```

**CardBuilder**
```kotlin
@Composable
fun CardBuilder(
  name: TextFieldValue,
```

then just pass it all the way down

Reply:
I don't think it is duplication, because for the DeckBuilder there is 2 textfield one for the name and the other for the description and for the BoosterPackBuilder there is only one textFiled for the name, I mainly followed this approach for the **KeyBoard options** it will be next then done for the DeckBuilder, and just done for the BoosterPackBuilder, and that's why I don't see it as duplication. but I will follow your approach

Reply:
That is true, but can be solved by passing the keyboard options down through CardBuilder.